components/decimal/src/options.rs

```diff
@@ -65,60 +52,3 @@ impl Default for GroupingStrategy {
         Self::Auto
     }
 }
-
-/// Configuration for when to render the minus sign or plus sign.
-///
-/// # Examples
-///
-/// ```
-/// use icu_decimal::options;
-/// use icu_decimal::FixedDecimalFormat;
-/// use icu_decimal::FormattedFixedDecimal;
-/// use icu_locid::Locale;
-/// use writeable::Writeable;
-///
-/// let locale = Locale::UND;
-/// let provider = icu_provider::inv::InvariantDataProvider;
-/// let mut options: options::FixedDecimalFormatOptions = Default::default();
-/// options.sign_display = options::SignDisplay::ExceptZero;
-/// let fdf = FixedDecimalFormat::try_new(locale, &provider, options)
-///     .expect("Data should load successfully");
-///
-/// let pos_thousand = 1000.into();
-/// assert_eq!("+1,000", fdf.format(&pos_thousand).write_to_string());
-///
-/// let zero = 0.into();
-/// assert_eq!("0", fdf.format(&zero).write_to_string());
-///
-/// let neg_thousand = (-1000).into();
-/// assert_eq!("-1,000", fdf.format(&neg_thousand).write_to_string());
-/// ```
-#[non_exhaustive]
-#[derive(Debug, Eq, PartialEq, Clone, Copy)]
-pub enum SignDisplay {
```

Comment:
Issue: I would like to retain the SignDisplay resolution algorithm code. I think it should go on `FixedDecimal`.

Reply:
What do you think about leaving it standalone, just changing the return type of `select` to `Sign`, so that you can `set_sign` on a `FixedDecimal` with the result?
The sign selection operation is part of the rounding operation 𝕽 in the diagram in https://docs.google.com/document/d/1yjLPwM08Y_gf6-3FhDI9uaB8t_OCjliRtUSoMJ9_r98/edit, which we don’t currently have—at least not in one place—; but once we get around to implementing it, I think it’s going to be clearer that way: you get the `Signum` of the input number, independently compute a `FixedDecimal` with the desired digits, and slap the displayed sign on at the end.

Reply:
`FixedDecimal` sits in an odd place where it is both the intermediate state and also a builder for the intermediate state.

To be most pure, we should have a `FixedDecimalBuilder` that collects the input number and all settings for it, like:

```rust
let mut builder = FixedDecimal::builder();
builder.set_i32(my_i32);
builder.set_sign_display(SignDisplay::ExceptZero);
builder.set_multiply_pow10(-2);
let fd = builder.build();
```

However, instead of using a builder, we just reify the decimal at each stage.

```rust
let mut fd = FixedDecimal::from(my_i32);
fd.apply_sign_display(SignDisplay::ExceptZero);
fd.multiply_pow10(-2);
```

Reply:
Works for me; done.